### PR TITLE
VC14 compatibility

### PR DESCRIPTION
**Pull request by Jan-E**
The change in ast-names.h was needed to solve this error:

```
c:\php-sdk\php70dev\ext\astkit\ast-names.h(36): error C2065: 'ZEND_AST_const':
undeclared identifier (compiling source file ext\astkit\astkit-node.c)
```

The other changes in the previous VC11 PR apparently aren't required anymore for VC14

### CODE REVIEW

**Comment by Jan-E** [comment]
@sgolemon ping

File: ast-names.h [hunk 1]
```diff
@@ -33,7 +33,7 @@ AST_CHILD(TYPE, 0)
 
 /* 1 child node */
 AST_CHILD(VAR, 1)
-AST_CHILD(CONST, 1)
+/* AST_CHILD(CONST, 1) */
```
Comment (sgolemon) [inline ast-names.h]: That sorta breaks any code which uses constant declarations.

Would it work if you put something like this at the top of the header?

```
#ifdef CONST
# undef CONST
#endif
```

**Comment by sgolemon** [comment]
Sorry I didn't see this PR until just now.  Feel free to poke me on twitter/SaraMG if I'm ever unresponsive.

**Comment by Jan-E** [comment]
@sgolemon Something must have been changed since the original PR. It does not throw an error anymore:

```
C:\php-sdk\php70dev>nmake php_astkit.dll

Microsoft (R) Program Maintenance Utility Version 14.00.23918.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        "cl.exe" /D COMPILE_DL_ASTKIT /D ASTKIT_EXPORTS=1 /nologo /I . /I main /I Zend /I TSRM /I ext /D _WINDOWS /D ZEND_WIN32=1 /D PHP_WIN32=1 /D WIN32 /D _MBCS /W3 /FD /wd4996 /D_USE_32BIT_TIME_T=1 /guard:cf /MP /Zi /LD /MD /W3 /Ox /D NDebug /D NDEBUG /D ZEND_WIN32_FORCE_INLINE /GF /D ZEND_DEBUG=0 /D ZTS=1 /I "C:\php-sdk\win32build\include" /D FD_SETSIZE=256/FoC:\php-sdk\php70dev\Release_TS\ext\astkit\ /FpC:\php-sdk\php70dev\Release_TS\ext\astkit\ /FRC:\php-sdk\php70dev\Release_TS\ext\astkit\ /FdC:\php-sdk\php70dev\Release_TS\ext\astkit\ /c ext\astkit\astkit-decl.c ext\astkit\astkit-list.c ext\astkit\astkit-node.c ext\astkit\astkit.c
astkit-decl.c
astkit-list.c
astkit-node.c
astkit.c
ext\astkit\astkit-list.c(25): warning C4018: '>=': signed/unsigned mismatch
ext\astkit\astkit-list.c(31): warning C4013: 'astkit_create_object' undefined; assuming ext
ern returning int
ext\astkit\astkit-decl.c(30): warning C4013: 'astkit_create_object' undefined; assuming extern returning int
C:\php-sdk\php70dev\Zend\zend_language_scanner.h(66): warning C4273: 'zend_compare_file_handles': inconsistent dll linkage (compiling source file ext\astkit\astkit-node.c)
c:\php-sdk\php70dev\zend\zend_stream.h(81): note: see previous definition of 'zend_compare_file_handles' (compiling source file ext\astkit\astkit-node.c)
ext\astkit\astkit-node.c(84): warning C4013: 'astkit_create_object' undefined; assuming extern returning int
Recreating build dirs
Recreating build dirs
Recreating build dirs
Recreating build dirs
Recreating build dirs
Recreating build dirs
libiconv_a.lib(iconv1.obj) : MSIL .netmodule or module compiled with /GL found; restarting link with /LTCG; add /LTCG to the link command line to improve linker performance
   Creating library C:\php-sdk\php70dev\Release_TS\php7ts.lib and object C:\php-sdk\php70dev\Release_TS\php7ts.exp
Generating code
Finished generating code
   Creating library C:\php-sdk\php70dev\Release_TS\php_astkit.lib and object C:\php-sdk\php70dev\Release_TS\php_astkit.exp
EXT astkit build complete
```

The 'inconsistent dll linkage' seems inherent to including zend_language_scanner.h. Nikita's AST extension has it as well.

**Comment by Jan-E** [comment]
BTW: The PR now only contains the config.w32

**Comment by sgolemon** [comment]
d7213a7da3a3ff35772ab36697da4a6a4e3c9211  should take care of the AstKit specific warnings, and it looks like @nikic already took care of the one related to zend_compare_file_handles()

**Pull request merged**